### PR TITLE
Tactical fix to update JWT validation parameters after specialization

### LINKED ISSUE #3174: JWT validation failure after specialization
JWT validation fails after a site goes through specialization. JWT validation parameters need to be reset.

### PR DESCRIPTION
**Pull request by fabiocav**
Resolves #3174

@pragnagopa this is a quick tactical fix to address the specialization issue. I'll create an item as I want to revisit this.

FYI @mathewc

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs [hunk 1]
```diff
@@ -18,11 +21,23 @@ namespace Microsoft.Extensions.DependencyInjection
...
+        private static double _specialized = 0;
+
         public static AuthenticationBuilder AddScriptJwtBearer(this AuthenticationBuilder builder)
             => builder.AddJwtBearer(o =>
                         {
                             o.Events = new JwtBearerEvents()
                             {
+                                OnMessageReceived = c =>
+                                {
+                                    // Temporary: Tactical fix to address specialization issues. This should likely be moved to a token validator
```
[the remark this answers is not in the document]
Comment (fabiocav) [reply src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs]: That hasn't been created yet. Will create

File: src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs [hunk 2]
```diff
@@ -18,11 +21,23 @@ namespace Microsoft.Extensions.DependencyInjection
...
+                                    if (_specialized == 0 && !WebScriptHostManager.InStandbyMode && Interlocked.CompareExchange(ref _specialized, 1, 0) == 0)
+                                    {
```
[the remark this answers is not in the document]
Comment (fabiocav) [reply src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs]: What do you mean? It is, in the `CompareExchange` call.

[the remark this answers is not in the document]
Comment (pragnagopa) [reply src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs]: Ahh. Thanks. Missed that. I was referring to the call below that was updating _specialized  to 1 if not in standby mode.

File: src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (fabiocav) [reply src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs]: By the way.... this is what fixes the issue. Before the change, the validation parameters were not reset after specialization, so we didn't have the proper site validation parameters/key.

**Pull request merged**